lib/kuzzle/event/KuzzleEventEmitter.ts

```diff
@@ -135,17 +157,21 @@ class KuzzleEventEmitter extends EventEmitter {
    * Emits an event and all its wildcarded versions
    *
    * @warning Critical section of code
-   *
-   * @param  {string} event
-   * @param  {*} data
    */
-  emit(event, data) {
+  emit(event: string | symbol, ...args: any[]): boolean {
     const events = getWildcardEvents(event);
-    debug('Triggering event "%s" with data: %o', event, data);
+    debug('Triggering event "%s" with data: %o', event, args);
+
+    if (events.length === 0) {
+      debug('No listeners for event "%s"', event);
+      return false;
+    }
 
-    for (let i = 0; i < events.length; i++) {
```

Comment:
This old for loop was choose to increase performance, I don't know if `for (let i = 0; i < ...)` is still faster than `for (const ... of ...)` in later Node.js versions